pkg/ccl/storageccl/export.go

```diff
@@ -91,85 +90,69 @@ func evalExport(
 		defer exportStore.Close()
 	}
 
-	sst, err := engine.MakeRocksDBSstFileWriter()
-	if err != nil {
-		return result.Result{}, err
-	}
-	defer sst.Close()
-
-	var skipTombstones bool
-	var iterFn func(*engineccl.MVCCIncrementalIterator)
+	var exportAllRevisions bool
 	switch args.MVCCFilter {
 	case roachpb.MVCCFilter_Latest:
-		skipTombstones = true
-		iterFn = (*engineccl.MVCCIncrementalIterator).NextKey
+		exportAllRevisions = false
 	case roachpb.MVCCFilter_All:
-		skipTombstones = false
-		iterFn = (*engineccl.MVCCIncrementalIterator).Next
+		exportAllRevisions = true
 	default:
 		return result.Result{}, errors.Errorf("unknown MVCC filter: %s", args.MVCCFilter)
 	}
 
-	debugLog := log.V(3)
+	start := engine.MVCCKey{Key: args.Key, Timestamp: args.StartTime}
+	end := engine.MVCCKey{Key: args.EndKey, Timestamp: h.Timestamp}
 
-	var rows bulk.RowCounter
-	// TODO(dan): Move all this iteration into cpp to avoid the cgo calls.
-	// TODO(dan): Consider checking ctx periodically during the MVCCIterate call.
-	iter := engineccl.NewMVCCIncrementalIterator(batch, engineccl.IterOptions{
```

Comment:
Suggestion: you could pull this code into a go function that takes the same args are your new c++ function (and keep mvcc.go around for now). Then you could write a unit test that calls both functions on the same engine and compares the result?

Reply:
Done.

Reply:
Done.